pyccel/parser/semantic.py

```diff
@@ -2364,7 +2364,7 @@ def _visit_Return(self, expr, **settings):
                 a.set_fst(expr.fst)
                 a = self._visit_Assign(a)
                 ind = a.rhs.get_attribute_nodes(IndexedElement)
-                if len(ind) == 0 and not isinstance(a.rhs, IndexedElement):
+                if len(ind) == 0 and not isinstance(a.rhs, IndexedElement) or isinstance(a.rhs, IndexedElement) and a.rhs.base.is_argument:
```

Comment:
hmm, are you sure about this ? 
did you try to test something like this ? when the rhs is not and indexed Element but it contain indexed Element of an argument example PyccelAdd 
``` PYTHON
@types('int[:]', 'int[:]')
def test(a, b):
        return a[0] + b[0]
```
result : 
```C
#include "avoid.h"
#include <stdint.h>
#include <stdlib.h>
#include "ndarrays.h"

/*........................................*/
int64_t test(t_ndarray a, t_ndarray b)
{
    int64_t Out_0001;
    return a.nd_int64[get_index(a, 0)] + b.nd_int64[get_index(b, 0)];
}
```

Reply:
I am also unsure about this. What about a case without an `IndexedElement`
```
def sum(a : 'int[:]'):
    total = 0
    for i in range(a.shape[0]):
        total += a[i]
    return total

def sum_to_n(n : 'int'):
    import numpy as np
    c = np.arange(1,b+1)
    return sum(c)
```